### Importación de datos



In [13]:
import pandas as pd

url1 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda1 = pd.read_csv(url1)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

tienda.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  tienda  
0  Tarjeta de crédito                   8   4.60971 -74.08175  tienda  
1  Tarjeta de crédito                   4   6.25184 -75.56359  tienda  
2  Tarjeta de crédito                   1  10.39972 -75.51444  tienda  
3               Nequi                   1   3.43722 -76.52250  tienda  
4               Nequi                   1   6.25184 -75.56359  tienda

# 1. Análisis de facturación



In [14]:
tienda1["tienda"] = "tienda1"
tienda2["tienda"] = "tienda2"
tienda3["tienda"] = "tienda3"
tienda4["tienda"] = "tienda4"

# Unir todas las tiendas
df_total = pd.concat([tienda1, tienda2, tienda3, tienda4], ignore_index=True)

In [ ]:

df_total.head()

Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  tienda  
0  Tarjeta de crédito                   8   4.60971 -74.08175  tienda  
1  Tarjeta de crédito                   4   6.25184 -75.56359  tienda  
2  Tarjeta de crédito                   1  10.39972 -75.51444  tienda  
3               Nequi                   1   3.43722 -76.52250  tienda  
4               Nequi                   1   6.25184 -75.56359  tienda

In [19]:
df_total["facturacion"] = df_total["Precio"] * df_total["Cantidad de cuotas"]
facturacion_total = df_total["facturacion"].sum()
print("Facturación total:", round(facturacion_total, 2))

facturacion_por_tienda = df_total.groupby("tienda")["facturacion"].sum().reset_index()
facturacion_por_tienda["facturacion"] = facturacion_por_tienda["facturacion"].apply(lambda x: f"{x:,.2f}")
print(facturacion_por_tienda)


Facturación total: 13102551000.0
    tienda       facturacion
0  tienda1  3,422,717,700.00
1  tienda2  3,337,189,200.00
2  tienda3  3,243,816,900.00
3  tienda4  3,098,827,200.00


# 2. Ventas por categoría

In [24]:

ventas_por_categoria = df_total.groupby("Categoría del Producto")["facturacion"].sum().reset_index()

ventas_por_categoria["facturacion"] = ventas_por_categoria["facturacion"].apply(lambda x: f"{x:,.2f}")

print("Ventas por categoría:")
print(ventas_por_categoria)

Ventas por categoría:
    Categoría del Producto       facturacion
0  Artículos para el hogar    170,360,400.00
1     Deportes y diversión    432,005,500.00
2        Electrodomésticos  4,019,072,300.00
3             Electrónicos  4,889,815,300.00
4   Instrumentos musicales  1,044,612,400.00
5                 Juguetes    211,615,600.00
6                   Libros    113,088,700.00
7                  Muebles  2,221,980,800.00


# 3. Calificación promedio de la tienda


In [ ]:
# Calcular calificación promedio por tienda
calificacion_promedio = df_total.groupby("tienda")["Calificación"].mean().reset_index()

calificacion_promedio["Calificación"] = calificacion_promedio["Calificación"].round(2)

print("Calificación promedio por tienda:")
print(calificacion_promedio)


Calificación promedio por tienda:
    tienda  Calificación
0  tienda1          3.98
1  tienda2          4.04
2  tienda3          4.05
3  tienda4          4.00


# 4. Productos más y menos vendidos

In [28]:
# Agrupar por producto y sumar la cantidad de cuotas (ventas)
ventas_por_producto = df_total.groupby("Producto")["Cantidad de cuotas"].sum().reset_index()

# Ordenar de mayor a menor
ventas_ordenadas = ventas_por_producto.sort_values(by="Cantidad de cuotas", ascending=False)

# Mostrar el top 5 más vendidos
print("Productos más vendidos:")
print(ventas_ordenadas.head())

# Mostrar el top 5 menos vendidos
print("\nProductos menos vendidos:")
print(ventas_ordenadas.tail())


Productos más vendidos:
            Producto  Cantidad de cuotas
40  Secadora de ropa                 649
32     Mesa de noche                 627
38         Pandereta                 625
8          Bicicleta                 614
25         Iphone 15                 611

Productos menos vendidos:
          Producto  Cantidad de cuotas
30  Mesa de centro                 469
19  Dinosaurio Rex                 458
13    Celular ABXY                 447
42    Set de vasos                 435
46      Smartwatch                 432


# 5. Envío promedio por tienda

In [29]:
envio_promedio_por_tienda = df_total.groupby("tienda")["Costo de envío"].mean().reset_index()

# Redondear a 2 decimales
envio_promedio_por_tienda["Costo de envío"] = envio_promedio_por_tienda["Costo de envío"].round(2)

print("Costo de envío promedio por tienda:")
print(envio_promedio_por_tienda)


Costo de envío promedio por tienda:
    tienda  Costo de envío
0  tienda1        26018.61
1  tienda2        25216.24
2  tienda3        24805.68
3  tienda4        23459.46


# Análisis final

* La tienda con mayor facturación es Tienda 1, con un total de 3,422,717,700.00, lo que evidencia un alto volumen de ventas o precios más elevados. En contraste, la tienda con menor facturación es Tienda 4, con 3,098,827,200.00, lo que podría deberse a una menor cantidad de productos vendidos o precios unitarios más bajos. 
* En cuanto a calificación promedio, la Tienda 3 tiene la mayor calificación con 4.05 puntos, y, la Tienda 1 tiene la menor calificación promedio con 3.98 puntos.
* El costo de Envio de la Tienda 1 es el mayor con respecto al resto, y quien ofrece menor costo de envio es la Tienda 4.

En síntesis: Se recomienda la venta de la Tienda 4 al realizar una comparación entre ellas con respecto a las ventas realizadas, a pesar de la baja calificación de la Tienda 1, este ofrece mayor facturación que otras tiendas y la Tienda 4 ofrece menos.